rewrite-java/src/main/java/org/openrewrite/java/internal/template/BlockStatementTemplateGenerator.java

```diff
@@ -264,7 +264,7 @@ private void template(Cursor cursor, J prior, StringBuilder before, StringBuilde
                 before.insert(0, "{\n");
             }
 
-            if (prior instanceof Statement) {
+            if (prior instanceof Statement && !(prior instanceof J.MethodDeclaration)) {
                 after.append(';');
             }
```

Comment:
`J.MethodDeclaration` and `J.ClassDeclaration` should not have a `;` appended, although it is generally not a syntax error to have one. (Does result in an extra empty statement being generated). 

There may be other exceptions to the rule, but those are the two that occur to me thinking about it.

Reply:
A number of control flow statements also come to mind as well (if, for, while, and switch to name a few)

Reply:
I was the one to add a semicolon after statements here. The problem as I remember it has to do with the fact that we have some expressions which can also be used as statements (e.g. method invocations) and unlike the Javac AST we don't wrap them in some kind of `ExpressionStatement` element.

I will take a look at this, add some more tests and possibly add a cleaner fix.